crates/builder/src/lib.rs

```diff
@@ -320,6 +411,29 @@ pub trait DirectedNeighborsWithValues<NI: Idx, EV> {
     fn in_neighbors_with_values(&self, node: NI) -> Self::NeighborsIterator<'_>;
 }
 
+/// Allows adding new edges to a graph.
+pub trait EdgeMutation<NI: Idx> {
+    /// Adds a new edge between the given source and target node.
+    ///
+    /// # Errors
+    ///
+    /// If either the source or the target node does not exist,
+    /// the method will return [`Error::MissingNode`].
```

Comment:
I was prepared to comment on adding an Errors sections about what happens when yadayada, but you've already covered everything :)